typify-impl/src/convert.rs

```diff
@@ -651,34 +668,64 @@ impl TypeSpace {
             (None, None) => None,
         };
 
-        // TODO we should do something with `multiple`
-        if let Some(ty) = maybe_type {
-            Ok((TypeEntry::new_integer(ty), metadata))
-        } else {
-            // TODO we could construct a type that itself enforces the various
-            // bounds.
-            // TODO failing that, we should find the type that most tightly
-            // matches these bounds.
-            Ok((TypeEntry::new_integer("i64"), metadata))
-        }
-    }
+        match (min, max, multiple) {
+            (None, None, None) => {
+                return Ok(if is_int {
+                    (
+                        TypeEntry::new_integer(if let Some(ty) = maybe_type {
+                            ty
+                        } else {
+                            "i64".to_string()
+                        }),
+                        metadata,
+                    )
+                } else {
+                    (
+                        TypeEntry::new_float(if let Some(ty) = maybe_type {
+                            ty
+                        } else {
+                            "f64".to_string()
+                        }),
+                        metadata,
+                    )
+                });
+            }
+            _ => {
+                let type_entry = if is_int {
+                    TypeEntryDetails::Integer(if let Some(ty) = maybe_type {
+                        ty
+                    } else {
+                        "i64".to_string()
+                    })
+                    .into()
+                } else {
+                    TypeEntryDetails::Integer(if let Some(ty) = maybe_type {
```

Comment:
This part feels a bit messy, like boilerplate wise, wasn't really sure how to deal with it well. Wish rust had something like `maybe_type ?? "i64".to_string()` for options